src/Language/Fortran/Parser/Fortran90.y

```diff
@@ -308,6 +312,94 @@ BLOCK :: { Block A0 }
   { BlInterface () (getTransSpan $1 $8) $2 False [ ] (reverse $5) }
 | COMMENT_BLOCK { $1 }
 
+IF_BLOCK :: { Block A0 }
+IF_BLOCK
+: if '(' EXPRESSION ')' then NEWLINE BLOCKS ELSE_BLOCKS
```

Comment:
As a side note, I tried to simplify this rule by incorporating `[label] [id :] if` into its own rule, but then if `StIfLogical` took precedence for some reason and it failed pars when it got to `then`. Any idea why/how to get around this?